Signal/src/util/ThreadUtil.m

```diff
@@ -239,18 +239,40 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
             // Due to disappearing messages or manual deletion,
             // firstUnseenInteractionTimestampParameter may refer to an obsolete
             // interaction in which case we want to discard it.
-            TSInteraction *_Nullable lastInteraction =
-                [[transaction ext:TSMessageDatabaseViewExtensionName] lastObjectInGroup:thread.uniqueId];
-            if (!lastInteraction
-                || lastInteraction.timestampForSorting
-                    < firstUnseenInteractionTimestampParameter.unsignedLongLongValue) {
+            //
+            // Therefore, we should discard the existing unread indicator
+            // position if there are no "unreadable" messages after it.
+            __block TSInteraction *lastCallOrMessage = nil;
+            [[transaction ext:TSMessageDatabaseViewExtensionName]
```

Comment:
This search should be fairly quick - the worst case is if there is an obsolete unread indicator followed by hundreds of error or info messages.  That's extremely unlikely.